lib/active_merchant/billing/gateways/cams.rb

```diff
@@ -0,0 +1,241 @@
+        '801' => STANDARD_ERROR_CODE[:processing_error],
+        '811' => STANDARD_ERROR_CODE[:processing_error],
+        '812' => STANDARD_ERROR_CODE[:processing_error],
+        '813' => STANDARD_ERROR_CODE[:processing_error],
+        '814' => STANDARD_ERROR_CODE[:processing_error],
+        '815' => STANDARD_ERROR_CODE[:processing_error],
+        '823' => STANDARD_ERROR_CODE[:processing_error],
+        '824' => STANDARD_ERROR_CODE[:processing_error],
+        '881' => STANDARD_ERROR_CODE[:processing_error],
+        '882' => STANDARD_ERROR_CODE[:processing_error],
+        '883' => STANDARD_ERROR_CODE[:processing_error],
+        '884' => STANDARD_ERROR_CODE[:card_declined], 
+        '885' => STANDARD_ERROR_CODE[:card_declined],
+        '886' => STANDARD_ERROR_CODE[:card_declined],
+        '887' => STANDARD_ERROR_CODE[:processing_error],
+        '888' => STANDARD_ERROR_CODE[:processing_error],
+        '889' => STANDARD_ERROR_CODE[:processing_error],
+        '890' => STANDARD_ERROR_CODE[:processing_error],
+        '891' => STANDARD_ERROR_CODE[:incorrect_cvc],
+        '892' => STANDARD_ERROR_CODE[:incorrect_cvc],
+        '893' => STANDARD_ERROR_CODE[:processing_error],
+        '894' => STANDARD_ERROR_CODE[:processing_error]
+      }
+
+      def initialize(options={})
+        requires!(options, :username, :password)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        post = {}
+        #transactionid
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+
+        commit("sale", post)
+      end
+
+      def authorize(money, payment, options={})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment)
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+
+        commit('auth', post)
+      end
+
+      def capture(money, authorization, options={})
+        transaction_id, authcode = split_authorization(authorization)
+        post = {
+          "transactionid" => transaction_id,
+          "authcode"      => authcode
+        }
+        add_invoice(post, money, options)
+        add_customer_data(post, options)
+
+        commit('capture', post)
+      end
+
+      def refund(money, authorization, options={})
+        transaction_id, authcode = split_authorization(authorization)
+        post = {
+          "transactionid" => transaction_id,
+          "authcode"      => authcode
+        }
+        add_invoice(post, money, options)
+        commit('refund', post)
+      end
+
+      def void(authorization, options={})
+        transaction_id, authcode = split_authorization(authorization)
+        post = {
+          "transactionid" => transaction_id,
+          "authcode"      => authcode
+        }
+        commit('void', post)
+      end
+
+      #def verify(credit_card, options={})
```

Comment:
With `authorize` and `void` being available, can you please implement `verify`? Even better if the gateway supports a zero dollar auth or similar.